---
layout: post
comments: true
title: Gigas Broodstock RNA Extraction
tags: manchester labwork gigas-broodstock RNA DNA
---

## *C. gigas* RNA and DNA extractions: Day 1

With everyone else in the lab extracting RNA, I figured I should too (#fomo...?). Over the next few days, I'll extract RNA and DNA from [frozen *C. gigas* tissues collected after 7 weeks of either low or ambient pH exposure in 2017](https://github.com/RobertsLab/project-oyster-oa/blob/master/data/Manchester/2017-Adult-Gigas-Tissue-Sampling/20170408-GigasTissueSamplingInformation.xlsx). These are the adults I [already extracted gonad DNA from](https://yaaminiv.github.io/Gigas-Broodstock-DNA-Extraction-Part9/). 

I followed [the ZymoResearch Quick DNA/RNA Microprep Plus Kit protocol](https://github.com/RobertsLab/resources/blob/master/protocols/Commercial_Protocols/ZymoResearch_quick-dna-rna_microprep_plus_kit_20190411.pdf), which is what [Grace uses for her crab samples](https://grace-ac.github.io/extract-RNA/). Since Grace has been able to use it with her trickier crab hemolymph samples, I'm more confident that it should work with standard frozen oyster tissues. The kit separates out DNA and RNA into a column and flow-through, so I should be able to get both molecules from one sample. Today I'll test out the protocol using adductor tissue samples from 20 individuals (10 low pH, 10 ambient pH). I decided to start with the adductor first since it's the tissue I care about least. It's not directly involved in any sort of ocean acidification acclimation processes, but could be affected. It might be a nice contrast to see how methylation, chromatin accessibility, and gene expression changes in relation to other tissues more active in ocean acidification acclimation like ctenidia and mantle tissues.

### Methods: Sample Preparation

#### Step 1: Prepare for extractions.

- Label 3 sets of tubes RNase-free centrifuge tubes per sample: one for frozen tissue, one for final RNA storage, and one for final DNA storage.
- Add 96 mL of 100% ethanol to the 24 mL DNA/RNA wash buffer concentrate. 
- Add 1040 µL Proteinase K Storage Buffer to Proteinase K vial. Vortex and store at -20ºC
- Set heat block to 55ºC
- Obtain samples from -80ºC freezer and place in ice: I did this before I did everything else since I was preoccupied with actually locating my samples. By the time I scrounged up the rest of my materials, they were more tissue than frozen...whoops. 

#### Step 2: Cut and weigh no 0.005 g (5 mg) of frozen tissue. Record weight of tissue used in extractions and place tissue in a new, labelled test tube.

- I tared the scale with a piece of weigh paper. I used tweezers to remove the tissue from the tube, then a razor blade to cut the tissue. Once I got the weight I wanted, I transferred the tissue to the labelled centrifuge tube with the tweezers.
- Tweezers and the razor blade were washed in 200 mL of a 10% bleach solution, then in two separate DI water rinses. I wiped the tools clean with a kim wipe to use again.
- I first tried using our lab's scale with samples 18 and 19, but I couldn't get a reading less than 10 mg even when barely any tissue. I borrowed the scale from Graham's lab and was able to get actual weights.

**Table 1**. Mass of samples used for RNA extractions.

| **Sample ID** | **Mass (mg)** |
|:-------------:|:-------------:|
|      1-T3     |       4.8     |
|      2-T1     |       4.7     |
|      3-T1     |       4.5     |
|      4-T3     |       5.1     |
|      5-T3     |       5.5     |
|      6-T1     |       5.3     |
|      7-T2     |       5.4     |
|      8-T2     |       5.1     |
|      9-T2     |       5.4     |
|     10-T3     |       5.3     |
|     11-T4     |       4.8     |
|     12-T6     |       4.4     |
|     13-T5     |       4.7     |
|     14-T6     |       4.8     |
|     15-T5     |       4.5     |
|     16-T4     |       5.5     |
|     17-T4     |       4.5     |
|     18-T6     |       4.6     |
|     19-T5     |       5.3     |
|     20-T6     |       5.0     |

#### Step 3: Add at least 150 µL of DNA/RNA shield (2X) and 150 µL nuclease-free water to create 300 µL DNA/RNA shield (1x) to each sample. If the sample is not covered by water, add more DNA/RNA shield (1x) until covered and record the volume of liquid added.

- I intially added 300 µL DNA/RNA shield (2x) to each sample, but I only noticed this after the samples were on the heat block...I removed samples from the heat block at 3 p.m. and added 300 µL nuclease free water to dilute the DNA/RNA shield.

#### Step 4: For every 300 µL of sample, add 30 µL PK Digestion Buffer and 15 µL Proteinase K. Mix by vortexing gently.

- After adding 300 µL nuclease-free water, my sample volume had effectively doubled. I added an additional 30 µL PK Digestion Buffer and 15 µL Proteinase K, totalling 60 µL PK Digestion Buffer and 30 µL Proteinase K per sample.

#### Step 5: Place samples on a heat block at 55ºC for 2-5 hours.

- Initially placed samples on the heat block at 1:55 p.m. At 3 p.m., I removed them from the heat block to correct mistakes in Step 3 and 4. I placed them back on the heat block at 3:20 p.m.

#### Step 6: Vortex sample and centrifuge at maximum speed for 2 minutes to pellet debris. Transfer the aqueous supernatant to an RNase-free tube.

#### Step 7: Add an equivalent volume of DNA/RNA Lysis Buffer to each sample and mix by vortexing.

### Methods: Sample Purification

#### Step 8: Transfer the sample into a IC-MX spin column in a collection tube and centrifuge at 15,000 x g for 30 seconds. 

- Keep the labelled spin columns for DNA extractions. I originally thought that I would extract RNA today and place the DNA column in the fridge for later extraction, but Sam told me that yields were really poor when he ketp samples in the fridge for a week. I decided to extract both nucleic acids today.
- Save the flow-through for RNA extractions
- Since I had to add extra liquid (s/o to past Yaamini for not internalizing directions), I had a lot of sample. Not all of the sample fit into the spin column. I spun the first 1000 µL (about 300 µL remaining) and processed the DNA columns first. When I removed the IC-MX columns with DNA from the collection tubes with flow-through and RNA, I completely spaced and forgot to label the collection tubes. By the time I went back to process them and elute RNA, I had no idea which sample was which! I had to discard the flow-through with RNA and tubes, but thankfully I still had sample remaining that had not gone through a collection tube. Good thing I'm making all these mistakes with low stakes samples that I have so much more of.

#### Step 9: For RNA only, add an equal volume of 95-100% ethanol to the flow-through and mix by pipetting. Transfer the flow-through into a new IC spin column in a clean collection tube. Centrifuge at 15,000 x g for 30 seconds. Discard the flow-through.

#### Step 10: Add 400 µL DNA/RNA Prep Buffer to the column. Centrifuge at 15,000 x g for 30 seconds. Discard the flow-through.

#### Step 11: Add 700 µL DNA/RNA Wash Buffer to the column. Centrifuge at 15,000 x g for 30 seconds. Discard the flow-through.

#### Step 12: Add 400 µL DNA/RNA Wash Buffer to the column. Centrifuge at 15,000 x g for 2 minutes. Transfer the column to a new microcentrifuge tube.

#### Step 13: Add 15 µL DNase/RNase-Free Water to the column. Incubate at room temperature for 5 minutes. Centrifuge at 15,000 x g for 30 seconds to elute the RNA.

- I stopped here and put my samples in the -80ºC to quantify the next morning.

*Updated 2020-02-05*

### Methods: Sample Quantification

#### Step 14: Prepare the master solution using a 1:200 ratio of buffer to dye. For each sample and both standards, 200 µL of master solution is needed.

- RNA: 6567 µL of buffer and 33 µL of dye using Qubit RNA HS kit
- DNA: I used the pre-mixed master solution for the Qubit dsDNA HS kit!!!!!!!!! Amazing

#### Step 14: For each standard’s designated Qubit assay tube, add 10 µL of the correct standard and 190 µL of the master solution.

#### Step 14: Vortex the sample before adding it to the assay tube. For each sample’s designated Qubit assay tube, add 1 µL of the sample and 199 µL of the master solution. 

- After vortexing I quickly spun down the tubes to get the liquid that splattered back down.

#### Step 14: Vortex all Qubit assay tube for 2-3 seconds at maximum speed. Incubate tubes at room temperature for two minutes.

#### Step 14: Use Qubit to quantify RNA and DNA yield in each sample tube.

### Results

**Table 2**. RNA concentration for extracted samples. A total of 15 µL of RNA or DNA was eluted per sample. Final RNA or DNA yields only consider the remaining 14 µL after using 1 µL for quantification.

| **Sample ID** | **RNA Concentration (ng/µL)** | **Total RNA Yield (ng)** | **DNA Concentration (ng/µL)** | **Total DNA Yield (ng)** |
|:-------------:|:-----------------------------:|:------------------------:|:-----------------------------:|:------------------------:|
|      1-T3     |               10              |            140           |              1.10             |           15.4           |
|      2-T1     |              8.86             |          124.04          |              1.14             |           15.96          |
|      3-T1     |              6.66             |           93.24          |             0.424             |           5.936          |
|      4-T3     |              12.4             |           173.6          |             0.660             |           9.24           |
|      5-T3     |              10.2             |           142.8          |              1.36             |           19.04          |
|      6-T1     |              12.3             |           172.2          |             0.344             |           4.816          |
|      7-T2     |              9.88             |          138.832         |             0.646             |           9.044          |
|      8-T2     |              5.84             |           81.76          |             0.544             |           7.616          |
|      9-T2     |              5.42             |           75.88          |             0.422             |           5.908          |
|     10-T3     |              5.34             |           74.76          |             0.896             |          12.544          |
|     11-T4     |              N/A              |            N/A           |             0.322             |           4.508          |
|     12-T6     |              N/A              |            N/A           |             0.130             |           1.82           |
|     13-T5     |              4.6              |           64.4           |             0.704             |           9.856          |
|     14-T6     |              6.8              |           95.2           |             0.960             |           13.44          |
|     15-T5     |              10.4             |           145.6          |             0.738             |          10.332          |
|     16-T4     |              4.0              |            56            |             0.908             |          12.712          |
|     17-T4     |              13.0             |            182           |             0.512             |           7.168          |
|     18-T6     |              7.2              |           100.8          |              2.60             |           36.4           |
|     19-T5     |              12.5             |            175           |             0.948             |          13.272          |
|     20-T6     |              5.6              |           78.4           |             0.372             |           5.208          |

Not bad for messing up twice! Even assuming my yield would double had I not made those mistakes, I still don't have enough for the 1000 ng minimum necessary to sequence each sample individually. I'd need to repeat extraction 4 more times for each sample for RNA, and DNA is completely out of the question.

Two thoughts:

- I shouldn't use this kit to extract DNA, period. I probably have a better chance getting a higher yield with the E.Z.N.A Mollusc Kit. 
- Since my input is only 5 mg, it may be a better use of my time to use a different RNA extraction kit that allows for a higher input volume. Laura used RNAzol for her extractions, which has a maximum tissue input of 100 mg. The only downside to using RNAzol is that it uses a phase separation which I know I am more likely to screw up. The RNeasy kit has a maximum tissue input of 30 mg and involves spin columns instead.

I'll consult with Steven and Sam to figure out my plan of action before doing any further extractions.

### Going forward

1. Figure out the best extraction option
2. Re-extract adductor RNA
3. Extract ctenidia RNA
4. Extract mantle RNA

{% if page.comments %}

<div id="disqus_thread"></div>
<script>

/**
*  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
*  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
/*
var disqus_config = function () {
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
};
*/
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://the-responsible-grad-student.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>

{% endif %}

<script id="dsq-count-scr" src="//the-responsible-grad-student.disqus.com/count.js" async></script>
