---
layout: post
comments: true
title: MethCompare Part 11
tags: MethCompare bedtools
---

## More feature tracks

My larger goal this week is to run [various CpG categories](https://github.com/hputnam/Meth_Compare/issues/61) and [DMC](https://github.com/hputnam/Meth_Compare/issues/63) through my genome feature characterization pipeline. Before doing this, I wanted to obtain [flanking regions](https://github.com/hputnam/Meth_Compare/issues/57) and [transposable element](https://github.com/hputnam/Meth_Compare/issues/56) tracks and [add them to my pipeline](https://github.com/hputnam/Meth_Compare/issues/59).

### Creating flanking regions

The easiest thing on my to-do list was to generate flanking region tracks. I could easily do this with `flankBed`, then use `subtractBed` to remove any overlap with annotated genes. My one hang-up was creating a genome information file, with a column for chromosome and another column with the length of the chromosome. To make this file for *C. virginica*, I simply looked at the NCBI website and copied down the information for the 11 chromosomes that existed. It wasn't going to be that simple for *M. capitata* or *P. acuta*.

Before posting an issue, I dug through the [FASTA Markdown](https://github.com/RobertsLab/code/blob/master/fasta.md) in the [code repository](https://github.com/RobertsLab/code). I found some truly FASTAstic code (lol) to obtain sequence lengths from a FASTA file using `awk`. In [this Jupyter notebook](https://github.com/hputnam/Meth_Compare/blob/master/scripts/Generating-Genome-Feature-Tracks.ipynb), I tested the code with the *M. capitata* genome FASTA file:

```
#Obtain sequence lengths for each chromosome
!awk '$0 ~ ">" {print c; c=0;printf substr($0,2,100) "\t"; } $0 !~ ">" {c+=length($0);} END { print c; }' \
Mcap.genome_assembly.fa \
> Mcap.genome_assembly-sequence-lengths.txt
```

I checked the end of both the sequence length file and the genome to make sure that the last chromosome in each file matched. I repeated this process with *P. acuta*, and saved the *M. capita* file [here](https://github.com/hputnam/Meth_Compare/blob/master/genome-feature-files/Mcap.genome_assembly-sequence-lengths.txt) and *P. acuta* file [here](https://github.com/hputnam/Meth_Compare/blob/master/genome-feature-files/Pact.genome_assembly-sequence-lengths.txt).

I used these files as input for my `bedtools` code. To create the flanking regions, I specified that I wanted to create 1000 kb flanks on either side of annotated genes with `flankBed`. I then piped the output into `subtractBed` and subtracted gene regions from my flanking regions. I did this in case there were any genes that overlapped with my flanking regions:

```
#Create 1kb flanking regions
#Subtract existing genes so flanks do not have any overlap
!{bedtoolsDirectory}flankBed \
-i Pact.GFFannotation.Genes.gff \
-g Pact.genome_assembly-sequence-lengths.txt \
-b 1000 \
| {bedtoolsDirectory}subtractBed \
-a - \
-b Pact.GFFannotation.Genes.gff \
> Pact.GFFannotation.flanks.gff
```

I saved the [*M. capitata*](https://github.com/hputnam/Meth_Compare/blob/master/genome-feature-files/Mcap.GFFannotation.flanks.gff) and [*P. acuta*](https://github.com/hputnam/Meth_Compare/blob/master/genome-feature-files/Pact.GFFannotation.flanks.gff) flanking region tracks. I also added these tracks to the [*M. capitata*](https://github.com/hputnam/Meth_Compare/blob/master/genome-feature-files/Mcap-Genome-Feature-Tracks.xml) and [*P. acuta*](https://github.com/hputnam/Meth_Compare/blob/master/genome-feature-files/Pact-Genome-Feature-Tracks.xml) IGV sessions with the other feature tracks.

![Screen Shot 2020-05-05 at 2 46 20 PM](https://user-images.githubusercontent.com/22335838/81119211-30744980-8edf-11ea-90ef-0743a736f9a9.png)

**Figure 1**. *M. capitata* genome features

![Screen Shot 2020-05-05 at 2 48 15 PM](https://user-images.githubusercontent.com/22335838/81119362-75987b80-8edf-11ea-90f0-d89e2fce74ef.png)

**Figure 2**. *P. acuta* genome features

### Intersections with flanking regions

In [this script](https://github.com/hputnam/Meth_Compare/blob/master/scripts/Characterizing-CpG-Methylation-5x.ipynb) I characterized the overlap between flanking regions and individual *M. capitata* and *P. acuta* samples. Before doing this, I went through my local repository and deleted all of the intermediate files generated by this script. Since I can download bedgraphs from `gannet`, produce the intermediate files using the code in the script, and save any relevant count information in a text file, I didn't need to host them in the repository. After removing files and adding them to the `gitignore`, I used `intersectBed` to characterize overlaps between the samples and flanking regions. I didn't split this up by upstream or downstream flanks since I figured just knowing about the flanks is good for now:

- [*M. capitata* overlaps with flanks](https://github.com/hputnam/Meth_Compare/blob/master/analyses/Characterizing-CpG-Methylation-5x/Mcap/Mcap-5x-mcFlanks-counts.txt)
- [*P. acuta overlaps with flanks](https://github.com/hputnam/Meth_Compare/blob/master/analyses/Characterizing-CpG-Methylation-5x/Pact/Pact-5x-paFlanks-counts.txt)

I also looked at the intersection between flanks and CG motifs. Flanking regions in *M. capitata* overlapped with 9.6% of CG motifs, and 17.8% of CG motifs in *P. acuta*.

### Going forward

4. Update methods
5. Update results
6. Redo CpG distributions with union bedgraphs
2. [Locate TE tracks](https://github.com/hputnam/Meth_Compare/issues/56)
2. [Characterize intersections between data and TE, and create summary tables](https://github.com/hputnam/Meth_Compare/issues/59)
3. [Create figures for CpG characterization in various genome features](https://github.com/hputnam/Meth_Compare/issues/60)
3. Figure out methylation island analysis

{% if page.comments %}

<div id="disqus_thread"></div>
<script>

/**
*  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
*  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
/*
var disqus_config = function () {
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
};
*/
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://the-responsible-grad-student.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>

{% endif %}

<script id="dsq-count-scr" src="//the-responsible-grad-student.disqus.com/count.js" async></script>
