---
layout: post
comments: true
title: CEABiGR
tags: virginica mixOmics sPLS CEABiGR
---

## Exon methylation and expression

I'm continuing the work we started at CEABiGR by playing around with an [R package called `mixOmics`](https://mixomicsteam.github.io/Bookdown/index.html). My goal is to conduct an sparse Partial Least Squares (sPLS) analysis with exon expression and methylation data to understand correlations between the datasets, and identify individual exons with expression or methylation patterns driving the correlations. I'm working in [this R Markdown script](https://github.com/sr320/ceabigr/wiki) and following the code from [this `mixOmics` handbook section](https://mixomicsteam.github.io/Bookdown/pls.html#biological-question-3).

### Filtering data

Based on the sPLS trials I did at FHL, I knew I needed to filter the expression and methylation data not just for the statistical element, but because I couldn't run the package with such large datasets! I started by filtering out missing values. If any sample was missing data for a specific exon, then that exon was removed from the dataframe. This wasn't enough to be able to run the analysis. Similar to what Katherine did for the [mQTL analysis](https://github.com/sr320/ceabigr/blob/main/code/mqtl_analysis.Rmd), I removed exons that were not variable between sample by finding the range of percent methylation for an exon, and removing it from the dataset if the range < 10% methylated. This is the same cutoff we used to define lowly methylated loci.

```
#Transpose methylation data
#Remove rows with missing data
#Rowwise range operation
#Remove rows without big differences in exon methylation
exonMethylationFilt <- exonMethylationMod %>%
  t(.) %>%
  as.data.frame(.) %>%
  rownames_to_column("exon") %>%
  drop_na(.) %>%
  rowwise() %>%
  mutate(range = max(c_across(where(is.numeric))) - min(c_across(where(is.numeric)))) %>%
  column_to_rownames("exon") %>%
  filter(., range > 10) %>%
  select(., !range) %>%
  t(.) %>%
  as.data.frame(.)
head(exonMethylationFilt) #Confirm formatting
sum(is.na(exonMethylationFilt)) #Confirm there are no NAs
ncol(exonMethylationFilt) #Number of exons remaining
```

For the [eQTL analysis](https://github.com/sr320/ceabigr/blob/main/code/eqtl_analysis.Rmd), Laura removed low frequency genes (sum of expression data < 10). Instead of using this cutoff, I plotted an interactive histogram of the range in expression values across all samples for a specific exon with `plot_ly`:

```
exonExpressionHist <- exonExpressionMod %>%
  t(.) %>%
  as.data.frame(.) %>%
  drop_na(.) %>%
  rowwise() %>%
  summarise(range = max(c_across(where(is.numeric))) - min(c_across(where(is.numeric)))) #Calculate the range of exon expression values across each sample

plot_ly(x = exonExpressionHist$range, type = "histogram") #Interactive plot of exon expression value ranges
```

I zoomed into the plot to identify how many counts were in the lowest bins: 0-20, 20-40, and 40-60. Based on these counts, I decided to remove exons with range values < 20. P.S. `plot_ly` is super cool for exploratory analyses!

![Screen Shot 2022-03-14 at 5 45 15 PM](https://user-images.githubusercontent.com/22335838/158266574-46ab227f-5ae8-4158-ac38-e2bdbaf118fc.png)

```
#Transpose expression data
#Remove rows with missing data
#Rowwise range operation
#Remove rows without big differences in exon expression
exonExpressionFilt <- exonExpressionMod %>%
  t(.) %>%
  as.data.frame(.) %>%
  rownames_to_column("exon") %>%
  drop_na(.) %>%
  rowwise() %>%
  mutate(range = max(c_across(where(is.numeric))) - min(c_across(where(is.numeric)))) %>%
  column_to_rownames("exon") %>%
  filter(., range > 20) %>%
  select(., !range) %>%
  t(.) %>%
  as.data.frame(.)
head(exonExpressionFilt) #Confirm formatting
sum(is.na(exonExpressionFilt)) #Confirm there are no NAs
ncol(exonExpressionFilt) #Number of exons remaining
```

### Full sample sPLS

I ran the default sPLS parameters with the filtered exon expression and methylation datasets:

```
exonResult.spls <- spls(X = exonMethylationFilt, Y = exonExpressionFilt,
                        keepX = c(25, 25), keepY = c(25,25)) #Run SPLS
```

![Screen Shot 2022-03-15 at 9 06 57 AM](https://user-images.githubusercontent.com/22335838/158384344-d88e7e1a-d46a-4680-a06c-0a964bc2b914.png)

![Screen Shot 2022-03-15 at 9 07 03 AM](https://user-images.githubusercontent.com/22335838/158384346-ae5e2458-ba34-477b-8c6b-954590176a8b.png)

**Figures 1-2**. sPLS sample plots for all samples.

Based on the subset I ran prior to filtration during the workshop, I suspected I needed to separate male and female data. The plots confirm this, with male and female data separating very clearly. Samples 16F and 52F are also separating from the rest of the female samples, which is interesting and may be something to deal with in the future. There's some slight separation of low and ambient pH samples, which I suspect will be clearer when I run sex-specific analyses. Now that I've run the full sample sPLS, I have the evidence I need to separate samples by sex!

### Going forward

1. Revise with new expression data from Ariana
2. Tune SPLS parameters
3. Run sex-specific SPLS
4. Identify drivers

{% if page.comments %}

<div id="disqus_thread"></div>
<script>

/**
*  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
*  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
/*
var disqus_config = function () {
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
};
*/
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://the-responsible-grad-student.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>

{% endif %}

<script id="dsq-count-scr" src="//the-responsible-grad-student.disqus.com/count.js" async></script>
