---
layout: post
comments: true
title: Cold Acclimation Green Crab Experiment Part 27
tags: green-crab-cold extractions PCR restriction-digest gel
---

## WA genotyping + some MA troubleshooting (because of course)

### Methods

**DNEasy Extraction**:

1. Set the benchtop incubator to 56°C.
2. Wipe down the bench surface, pipettes, and racks with RNase Away.
3. Take out and label one 1.5 mL tube for each sample.
4. Add 180 µL Buffer ATL to each sample tube.
5. One at a time, cut a small amount of tissue from each sample (~1mm x 1mm) and place in the corresponding labeled 1.5mL tube. Make sure to ethanol and flame tools in between samples. If the specimen is stored in ethanol, make sure to blot off all ethanol before adding tissue to the labeled tube with Buffer ATL.
6. Add 20 µL Proteinase K to each sample. Mix samples by vortexing and quick-spin tube(s).
7. Place tube(s) in incubator at 56°C until tissue is completely lysed. Ideally, the incubation will be overnight. If opting for an incubation that isn't overnight, set an initial timer for 30 minutes and vortex tube(s) every 10 minutes. If tissue appears to not be fully dissolved after 30 minutes, continue to incubate samples and vortex every 15 minutes until tissue has dissolved.
8. After incubation and complete lysis, vortex sample(s) for 15 seconds and quick-spin tube(s).
9. Add 200 µL Buffer AL to each sample. Mix thoroughly by vortexing and quick-spin tube(s).
10. Incubate sample(s) at 56°C for 10 minutes.
11. While samples are incubating, take out and label one new 1.5 mL tube for each sample.
12. Add 200 µL 200 Proof Ethanol to each sample. Mix thoroughly by vortexing and quick-spin tube(s).
13. Pipet the entire sample volume (600 µL) into a labeled DNeasy Mini Spin column placed in a 2 mL collection tube (supplied). 14. Centrifuge for 1 minute at ≥6000 xg (8000 rpm). Discard the flow-through and collection tube.
15. Place the spin column in a new 2 mL collection tube (supplied). Add 500µl Buffer AW1 to each sample. Centrifuge for 1 minute at ≥6000 xg (8000 rpm). Discard the flow-through and collection tube.
16. Place the spin column in a new 2 mL collection tube (supplied). Add 500 µL Buffer AW2 to each sample and centrifuge for 3 minutes at 20,000 xg (14,000 rpm). Discard the flow-through and collection tube.
17. Transfer each spin column to the correct labeled 1.5 mL tube (not supplied).
18. Elute the DNA by adding 60 µL Buffer AE directly to the center of each spin column membrane. Dispense Buffer AE with pipette tip as close to the membrane as possible, without touching the membrane with the tip. It is strongly recommended to use a new tip for each sample.
19. Incubate for 1 minute at room temperature (15°C - 25°C). Centrifuge for 1 minute at ≥ 6000 xg (8000 rpm).
20. Take the eluted 60 µL and add it back to the membrane. Incubate at room temperature for 1 minute then centrifuge for 1 minute at ≥6000 xg (8000 rpm).

**PCR**:

1. IF STARTING HERE: Use RNAse AWAY to clean bench space, pipets, tip boxes, etc.
2. Label either an 8-strip PCR tube or a 96-well plate with sample ID and "P" (ex. 3P) and a 1.5 mL eppendorf tube "PCR MM." Be sure to label an extra tube for the PCR blank. Label two tubes for a known CC and a known TT crab.
3. Calculate the amount of reagents needed for PCR master mix for samples, a PCR blank, and two extra reactions.
  - GoTaq: 12.5 µL/sample
  - F primer: 2.5 µL/sample
  - R primer: 2.5 µL/sample
  - DEPC H<sub>2</sub>O: 5.5 µL/sample
4. Preheat the thermocycler with the SMC_60RD PCR protocol
5. Get ice and thaw PCR reagents and DNA at room temperature. Briefly vortex, spin down, and move onto ice as soon as reagents thaw.
6. *IF NEEDED*, make new F and R primer aliquots. Take the 100 µM stock bottles (blue lids) and dilute into a new, labelled 1.5 mL eppendorf tube in a 1:10 dilution (ex. 10 µL primer stock, 90 µL NF water).
6. Make master mix based on calculations in step 3 in the labelled PCR MM tube. Vortex, spin down, and keep on ice.
7. Aliquot 23 µL of PCR MM from step 6 in each tube for the samples, extraction blank, and PCR blank.
8. Once the DNA thaws, briefly vortex DNA and spin down. Place on an ice block tube holder.
9. Add 2 µL of DNA into each sample tube. Add 2 µL DEPC water for the PCR blank.
10. Seal wells completely. Use tube/well caps, foil plate seal, or Microseal A film to seal plates. Use roller to push down seal and use the plate sealer tool to ensure all wells are completely sealed.
11. Vortex the sealed tubes and briefly spin down.
12. Place in the thermocycler, close lid, and run the SMC_60RD PCR protocol.
13. When protocol is complete, either place product in the 4ºC or proceed to the restriction digest.

**Restriction Digest**:

1. Preheat the thermocycler with the SMCRD_IN incubation protocol
2. Obtain Alul enzyme from the -20ºC. VERY gently vortex, spin down, and place on ice.
3. Add 0.5 µL enzyme to each PCR reaction. Pipet up and down to mix
4. Briefly spin down
5. Place in thermocycler, close lid, and run the SMCRD_IN protocol
6. When incubation is finished, either place product in the 4ºC or proceed to gel imaging.

**Gel**:

1. Obtain product for gel, ladder, and TriTrak from the 4ºC fridge and move into gel room.
2. Microwave pre-made gel mix in the microwave for 3 minutes in 1.5 minute intervals. Be sure to swirl the bottle to mix the gel liquid.
3. Allow the bottle with gel mix to cool for a few minutes. While cooling, tape off sides of the gel tray.
3. After bottle has cooled enough to the touch but the gel mix is still in liquid form, pour the mix into the taped gel tray slowly to avoid bubbles. Use combs to make wells by slowly inserting them into the gel to avoid bubbles.
4. Allow gel to harden for at least two hours.
5. Once gel is hardened, place in gel box with 1x TAE buffer. If an extra gel was made, slide and place in a Ziploc bag with TAE buffer. Place that Ziploc bag in the same drawer as the pre-made gel mix, away from the light.
6. Obtain a piece of parafilm and pipet 1 µL of TriTrak dye for each sample, extraction blank, and PCR blank. On the parafilm, the dye will bead up. Place the dots far enough apart to avoid contamination.
7. Take 10 µL of DNA, 6 µL of PCR product, or 20 µL of restriction digest product and mix with a 1 µL dot of TriTrak dye by pipetting up and down at least 10 times. Then, pipet up 6 µL and load gel. Repeat for each sample until halfway through samples. Then, add 3 µL of diluted ladder + dye mix, and continue with remaining samples.
8. Run gel for 65 minutes at 90 V.
9. After running gel, remove gel tray and image. If needed, run gel for longer in 5-10 minute intervals.

### 2024-12-09

Alright! I'm going to start another batch of WA extractions. This is the last batch for the ~30ish crabs that were alive at the end of the experiment, which gives me the most power to understand genotype x time interactions on TTR.

#### Notes

- Samples for DNEasy extraction: 154, 156, 161, 167, 175, 195, 196, 215, 217, 221, 223, 228, 235, 240, 246, 250, 251, 256, 259

### 2024-12-10

In addition to running a gel with the samples from yesterday, I'm including the [three supposed MA CC crabs that showed up as CT on my genotyping gel: 3, 19, and 118](https://yaaminiv.github.io/Green-Crab-Experiment-2024-Part26/).

#### Notes

- Finished DNEasy extractions for: 154, 156, 161, 167, 175, 195, 196, 215, 217, 221, 223, 228, 235, 240, 246, 250, 251, 256, 259
- Additional MA samples for PCR: 3, 19, 118
- PCR Master Mix calculations
  - GoTaq: 12.5 µL x 25 = 312.5 µL
  - F: 2.5 µL x 25 = 62.5 µL
  - R: 2.5 µL x 25 = 62.5 µL
  - NF H<sub>2</sub>: 5.5 µL x 25 = 137.5 µL
- Almost out of Alul! Promega discontinued theirs, but Carolyn got a free one from ThermoFisher.
- Made new ladder: 5 µL Gene Ruler, 5 µL dye, 35 µL NF H<sub>2</sub>
- Started DNEasy extractions for: 148, 153, 157, 160, 164, 171, 176, 180, 181, 192, 197, 199, 207, 210, 213, 220, 222, 224, 238, 239, 245, 254, 258, 262

#### Results

![Screenshot 2024-12-10 at 4 44 49 PM](https://github.com/user-attachments/assets/abdc1a32-89e1-49b6-855f-cd3b3bc97c70)

**Figure 1**. Restriction digest gel image

Well, those MA crabs are definitely CT. I'll need to pick a few more MA samples that have been genotyped as CC and re-genotype to see if all of my CCs are CT, or if just some are.

Apart from a few faint samples, looks like most WA crabs can be attributed to CT or TT! This makes sense, since we know there's less C in the Washington population. However, yesterday I was talking with Carolyn and she mentioned how some samples that appeared to be CT with a clear top band and a very faint and thin bottom band turned out to be TT when genotyped. I'm taking note of all my imbalanced CT genotypes with the goal of rerunning them. If they have a bright top band and much fainter bottom band, then I think I will need to send them to Sequegen to confirm their genotypes. I'll need to do this with the MA samples as well.

### 2024-12-12

Since I don't have much Alul left and I don't really want to do PCR today I'm just going to finish the DNEasy extraction I started yesterday. I'm also going to get a head start on re-extracting a few MA samples for troubleshooting.

#### Notes

- Finished DNEasy extractions for: 148, 153, 157, 160, 164, 171, 176, 180, 181, 192, 197, 199, 207, 210, 213, 220, 222, 224, 238, 239, 245, 254, 258, 262
  - After addition of 200 µL of 200 proof ethanol, I vortexed samples 207, 210, 213, 220, 222, and 224. I noticed that 224 wasn't fully shut when I took the tubes out of the mini centrifuge and there was some liquid on the centrifuge. I cleaned everything thoroughly with RNAse Away
  - So many broken caps from the centrifuge today?!
  - According to my genotype spreadsheet, sample 160 is already extracted! Confirmed in [this lab notebook post](https://yaaminiv.github.io/Green-Crab-Experiment-2024-Part26/)
- Pipetted 5 µL of Alul, meaning I could digest 10 more samples at least. Based on that, I decided to start extractions for some MA samples to confirm genotypes tomorrow
- Started DNEasy for MA crabs: 38, 109, 30, 94, 16, 116
  - [38 and 109 were genotyped 6/18/24](https://yaaminiv.github.io/Green-Crab-Experiment-2024-Part12/)
  - [30 and 94 were genotyped 6/21/24](https://yaaminiv.github.io/Green-Crab-Experiment-2024-Part12/)
  - [16 and 116 were genotyped 7/9/24](https://yaaminiv.github.io/Green-Crab-Experiment-2024-Part15/)

### 2024-12-13

I'm harnessing my "I was born on Friday the 13th" energy in lab today to hopefully get some clarity on the next steps with the MA samples! We got ThermoFisher Alul yesterday, but I'll run WA samples with that restriction enzyme after testing these MA samples.

#### Notes

- Finished DNEasy for MA crabs: 38, 109, 30, 94, 16, 116
  - 30 was slightly yellow after the overnight incubation
  - Used new Buffer AL
- PCR and RD for MA crabs: 99, 38, 109, 30, 94, 16, 116
  - [99 was extracted with DNEasy and genotyped on 11/22/24](https://yaaminiv.github.io/Green-Crab-Experiment-2024-Part24/), which is why I didn't need to re-extract it
- PCR Master Mix calculations
  - GoTaq: 12.5 µL x 10 = 125 µL
  - F: 2.5 µL x 10 = 25 µL
  - R: 2.5 µL x 10 = 25 µL
  - NF H<sub>2</sub>: 5.5 µL x 10 = 55 µL
  - Made new F primer

#### Results

![Screenshot 2024-12-13 at 3 15 30 PM](https://github.com/user-attachments/assets/bc541682-7356-457f-acfa-f808e5ccb389)

![Screenshot 2024-12-13 at 3 15 46 PM](https://github.com/user-attachments/assets/53553f8b-8a4a-4012-9faf-5e5357ee9d6d)

**Figures 2-3**. Restriction digest gel image and computer image

Well...I think I may have jinxed myself because those bands look weird. Pro: they all look like they're the same genotype, which increases the likelihood that they're CC. Cons: the bands are higher than I would expect for CC or CT. They're very smushed and there's even some discrepancies in the pattern of the top bad. On the computer, it looked like there was a very tiny separation in two bands, but without that distinct hourglass shape that's often seen with CT I'm not confident about. Like a fool I ran genotypes I thought were CC without any CT or TT to contrast.

I'm redo the PCR, digest, and gel with the same samples, but include a known CT and TT as contrasting samples to help with the genotyping.

### 2024-12-16

Alright, let's QC those MA samples again!

#### Notes

- PCR and RD for MA crabs: 99, 38, 109, 30, 94, 16, 116, 35 (CT), 100 (TT)
- PCR Master Mix calculations
  - GoTaq: 12.5 µL x 14 = 175 µL
  - F: 2.5 µL x 14 = 35 µL
  - R: 2.5 µL x 14 = 25 µL
  - NF H<sub>2</sub>: 5.5 µL x 14 = 77 µL
- Used new ThermoFisher Alul

#### Results

### Going forward

1. Troubleshoot QC gel discrepancies by rerunning MA CC samples
2. Run a subset of WA samples with the new Alul
2. Identify samples for confirmation sequencing (across Chelex and DNEasy extractions)
2. Individual-level TTR data analysis for MA samples
2. Genotype WA samples
2. Individual-level TTR data analysis for WA samples
2. Clarify methods for average TTR analysis (reach out to Andy, Nic, or Megan?)
3. Determine methods for comparing population responses
3. Troubleshoot lipid assay protocol
5. Conduct lipid assay for crabs of interest

{% if page.comments %}

<div id="disqus_thread"></div>
<script>

/**
*  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
*  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
/*
var disqus_config = function () {
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
};
*/
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://the-responsible-grad-student.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>

{% endif %}

<script id="dsq-count-scr" src="//the-responsible-grad-student.disqus.com/count.js" async></script>
