---
layout: post
comments: true
title: Cold Acclimation Green Crab Experiment Part 23
tags: green-crab-cold extractions PCR restriction-digest gel
---

## Returning to genotyping troubleshooting

I have enough time now to FINALLY figure out what's going on with my genotyping. I decided to do the following:

- Extract samples using our Chelex and Chelex from Mindy
- Run Qubit with extracted samples
- Run PCR product on a gel
- Run a restriction digest

### Methods

**Chelex Extraction**:

1. Use RNAse AWAY to clean bench space, pipets, tip boxes, etc.
2. Obtain samples in ethanol from the -80ºC freezer.
2. Obtain two sets of either an 8-strip PCR tube set or a 96-well plate. Label one set with sample ID, and another with sample ID and "S" (ex. 3 and 3S). Label an eppendorf tube "Chelex."
3. Preheat the thermal cycler (protocol CHELEX) or set up a thermal block to 100ºC
4. Prepare a 10% Chelex solution (ex. 0.1 g Chelex beads in 1 mL of DEPC/nuclease-free water). Vortex thoroughly (10-15 seconds) and spin down briefly (5 secconds)
5. Add 70 µL of Chelex solution to each tube. Vortex Chelex solution for 10-15 seconds in between each sample tube since the Chelex beads settle quickly
6. Obtain and set up a flaming station and two pairs of tweezers. Ethanol and flame tweezers, then place on a clean kim wipe.
7. For each sample, use one tweezer to remove the leg joint from the sample tube with ethanol, and another pair of tweezers to remove the tissue from the leg. Be sure to avoid any exoskeletal pieces, as the chiton in the shell can inhibit the Chelex reaction. Place the tissue on a clean kim wipe and press to dab ethanol from the sample (which can also impede the Chelex reaction). After placing the blotted tissue in the sample tube, ethanol, flame, and toss the kim wipe. **Use RNAse AWAY to clean bench**.
8. Repeat step 7, **ensuring a clean kim wipe is used to blot each sample after cleaning the bench with RNAse AWAY**.
9. If doing extractions in a plate, seal wells with caps.
10. Vortex samples for 10-15 seconds, and spin samples briefly (5-10 seconds).
11. Place samples in the thermal cycler and run the CHELEX protocol, or on a heat block for 20 minutes at 100ºC.
12. If doing extraction sin a plate, quickly spin down samples, remove caps, and add 50 µL DEPC water to each sample to assist with evaporation issues. Recover plate with a foil seal.
13. Vortex samples for 10-15 seconds, then spin down for 2 minutes. Let tubes sit an additional minute or two, if possible, for Chelex beads to settle and make pipetting easier.
14. Pipet ~50 µL supernatant into new, labelled tubes or a plate (labelled "S"). Avoid the Chelex beads when pipetting.
15. Place the supernatant with DNA in the dirty -20ºC freezer until ready for PCR.

**PCR**:

1. IF STARTING HERE: Use RNAse AWAY to clean bench space, pipets, tip boxes, etc.
2. Label either an 8-strip PCR tube or a 96-well plate with sample ID and "P" (ex. 3P) and a 1.5 mL eppendorf tube "PCR MM." Be sure to label an extra tube for the PCR blank. Label two tubes for a known CC and a known TT crab.
3. Calculate the amount of reagents needed for PCR master mix for samples, a PCR blank, and two extra reactions.
  - GoTaq: 12.5 µL/sample
  - F primer: 2.5 µL/sample
  - R primer: 2.5 µL/sample
  - DEPC H<sub>2</sub>O: 5.5 µL/sample
4. Preheat the thermocycler with the SMC_60RD PCR protocol
5. Get ice and thaw PCR reagents and DNA at room temperature. Briefly vortex, spin down, and move onto ice as soon as reagents thaw.
6. *IF NEEDED*, make new F and R primer aliquots. Take the 100 µM stock bottles (blue lids) and dilute into a new, labelled 1.5 mL eppendorf tube in a 1:10 dilution (ex. 10 µL primer stock, 90 µL NF water).
6. Make master mix based on calculations in step 3 in the labelled PCR MM tube. Vortex, spin down, and keep on ice.
7. Aliquot 23 µL of PCR MM from step 6 in each tube for the samples, extraction blank, and PCR blank.
8. Once the DNA thaws, briefly vortex DNA and spin down. Place on an ice block tube holder.
9. Add 2 µL of DNA into each sample tube. Add 2 µL DEPC water for the PCR blank.
10. Seal wells completely. Use tube/well caps, foil plate seal, or Microseal A film to seal plates. Use roller to push down seal and use the plate sealer tool to ensure all wells are completely sealed.
11. Vortex the sealed tubes and briefly spin down.
12. Place in the thermocycler, close lid, and run the SMC_60RD PCR protocol.
13. When protocol is complete, either place product in the 4ºC or proceed to the restriction digest.

**Restriction Digest**:

1. Preheat the thermocycler with the SMCRD_IN incubation protocol
2. Obtain Alul enzyme from the -20ºC. VERY gently vortex and place on ice.
3. Add 0.5 µL enzyme to each PCR reaction. Either pipet up and down to mix or VERY gently and briefly vortex
4. Briefly spin down
5. Place in thermocycler, close lid, and run the SMCRD_IN protocol
6. When incubation is finished, either place product in the 4ºC or proceed to gel imaging.

**Gel**:

1. Obtain product for gel, ladder, and TriTrak from the 4ºC fridge and move into gel room.
2. Microwave pre-made gel mix in the microwave for 3 minutes in 1.5 minute intervals. Be sure to swirl the bottle to mix the gel liquid.
3. Allow the bottle with gel mix to cool for a few minutes. While cooling, tape off sides of the gel tray.
3. After bottle has cooled enough to the touch but the gel mix is still in liquid form, pour the mix into the taped gel tray slowly to avoid bubbles. Use combs to make wells by slowly inserting them into the gel to avoid bubbles.
4. Allow gel to harden for at least two hours.
5. Once gel is hardened, place in gel box with 1x TAE buffer. If an extra gel was made, slide and place in a Ziploc bag with TAE buffer. Place that Ziploc bag in the same drawer as the pre-made gel mix, away from the light.
6. Obtain a piece of parafilm and pipet 1 µL of TriTrak dye for each sample, extraction blank, and PCR blank. On the parafilm, the dye will bead up. Place the dots far enough apart to avoid contamination.
7. Take 10 µL of DNA, 6 µL of PCR product, or 10 µL of restriction digest product and mix with a 1 µL dot of TriTrak dye by pipetting up and down at least 10 times. Then, pipet up 6 µL and load gel. Repeat for each sample until halfway through samples. Then, add 3 µL of diluted ladder + dye mix, and continue with remaining samples.
8. Run gel for 40 minutes.
9. After running gel, remove gel tray and image.

### 2024-11-12

#### Notes

- Used Chelex from Tepolt lab and HAB lab (s/o to Mindy)
- Samples for extractions, PCR, and digest: 3, 5, 19, 74, 115
- Qubit: 2600 µL of pre-mixed working solution for 1x dsDNA HS assay
- Used existing GoTaq aliquot, NF H<sub>2</sub> aliquot, and F and R primers. Opened new GoTaq and NF H<sub>2</sub> aliquots
- PCR Master Mix calculations
  - GoTaq: 12.5 µL x 15 = 187.5 µL
  - F: 2.5 µL x 15 = 37.5 µL
  - R: 2.5 µL x 15 = 70 µL
  - NF H<sub>2</sub>: 5.5 µL x 15 = 82.5 µL
- Used 6 µL of PCR product for a gel
- Added 0.5 µL of Alul to PCR product after the PCR product gel was run. Some samples may have received 1 µL of Alul because I lost track of which samples I completed.

#### Results

**Table 1**. Qubit results (ng/µL) for extracted samples. S1 = 58.56, S2 = 26571.55

| **Sample** | **Chelex Origin** | **Concentration (ng/µL)** |
|:----------:|:-----------------:|:-------------------------:|
|      5     |       Tepolt      |            2.34           |
|     19     |       Tepolt      |            2.23           |
|     74     |       Tepolt      |            1.65           |
|      3     |       Tepolt      |            1.88           |
|     115    |       Tepolt      |            1.78           |
|      5     |        HAB        |            1.78           |
|     19     |        HAB        |            5.39           |
|     74     |        HAB        |            3.09           |
|      3     |        HAB        |            1.49           |
|     115    |        HAB        |            1.93           |

Across the board I had really low yields! While subsampling the tissue, I noticed I was taking the lower end of tissue instead of my usual amount of tissue, so that could be a contributing factor. Interesting to see that the two samples with the highest yields are not with the Tepolt Lab Chelex.

<img width="636" alt="Screenshot 2024-11-13 at 11 17 24 AM" src="https://github.com/user-attachments/assets/14cc0078-8f16-4d72-ae09-544b2f80967e">

**Figure 1**. PCR product gel

I obviously had some samples with yields too low for bright bands, but I had some with a bright enough band intensity for Sanger sequencing!

<img width="611" alt="Screenshot 2024-11-13 at 11 17 37 AM" src="https://github.com/user-attachments/assets/a37d9df8-c9d2-4db2-abab-538adbe9186d">

**Figure 2**. Restriction digest gel

Given how the PCR bands weren't the strongest to begin with, I'm not surprised that the restriction digest bands aren't discernible. Whatever brightness exists with the PCR product gets halved because the bands are split! These two gels confirm to me that the issue isn't with PCR amplification or the digest, but with the input DNA. The other thing I realized afterwards is that I used 6 µL of restriction digest product for the gel when I was using 10 µL for brighter bands previously. My next step is to try doing Chelex extractions with the two different Chelex sources again, but with more tissue. I also want to run the DNA out on a gel to ensure the DNA isn't degraded.

It's still weird to me that all of a sudden Chelex extractions don't seem to be working! When I was troubleshooting this summer, I did run Chelex with [this year's WA crab samples](https://yaaminiv.github.io/Green-Crab-Experiment-2024-Part16/) and [last year's WA crab samples](https://yaaminiv.github.io/Green-Crab-Experiment-2024-Part15/) and still got faint bands. The fact that the samples from last year's experiments — which had bright enough bands for Sanger sequencing AND I used to develop this protocol — had faint or non-existent bands when I re-extracted to create more restriction digest product makes me think that the issue really is the Chelex. I at least know that the issue isn't isolated to samples from this year.

### 2024-11-13

Today the goal was to understand what was happening with the Chelex extractions! I used Tepolt lab and HAB lab Chelex on the same samples as yesterday, but with more tissue for the extractions.

#### Notes

- Used Chelex from Tepolt lab and HAB lab
- Samples for extractions, PCR, and digest: 3, 5, 19, 74, 115
- Qubit: 2600 µL of pre-mixed working solution for 1x dsDNA HS assay
- Used 10 µL of DNA for the gel
- PCR Master Mix calculations
  - GoTaq: 12.5 µL x 15 = 187.5 µL
  - F: 2.5 µL x 15 = 37.5 µL
  - R: 2.5 µL x 15 = 70 µL
  - NF H<sub>2</sub>: 5.5 µL x 15 = 82.5 µL

#### Results

**Table 1**. Qubit results (ng/µL) for extracted samples. S1 = 48.42, S2 = 26887.85

| **Sample** | **Chelex Origin** | **Concentration (ng/µL)** |
|:----------:|:-----------------:|:-------------------------:|
|      5     |       Tepolt      |            5.61           |
|     19     |       Tepolt      |            3.39           |
|     74     |       Tepolt      |            2.14           |
|      3     |       Tepolt      |            1.61           |
|     115    |       Tepolt      |            2.15           |
|      5     |        HAB        |            5.12           |
|     19     |        HAB        |            2.86           |
|     74     |        HAB        |            3.05           |
|      3     |        HAB        |            1.25           |
|     115    |        HAB        |            2.10           |

Better yields for almost all of my samples today! It's also interesting to see that there isn't a difference in total yield between the two Chelex resins.

<img width="626" alt="Screenshot 2024-11-13 at 11 50 53 AM" src="https://github.com/user-attachments/assets/3c43fc11-53dd-44a6-adb6-42593befd8bc">

**Figure 3**. DNA product gel

Looking at the gel, it's evident that there is a decent amount of degraded DNA in all samples. This tracks because it is Chelex: it's quick and efficient, but not the cleanest. Even for samples with similar yields across the two Chelex resins, there does seem to be a brighter smear for samples from the HAB lab Chelex. I even cropped the really bright corner out of the gel and rebalanced the colors and still saw that the HAB lab Chelex samples had brighter smears! After showing the gel image to Carolyn, I decided to proceed with PCR and the restriction digest just to see if using more tissue helped.

Some additional points:

- I haven't done a successful Chelex extraction since June 22!
- 10 µL of restriction digest product helps even some fainter samples show up on the gel well enough to genotype
- She requested a sample of Chelex 100-200 mesh. Might be worth requesting this on my end as well.
- The Chelex from Mindy is 200-400 mesh
- I have 58 more MA samples to genotype and all 104 WA samples left to genotype

### 2024-11-14

#### Notes

- Ran a restriction digest using the PCR product from yesterday
- Used 10 µL of restriction digest product for a gel

### Results

<img width="626" alt="Screenshot 2024-11-14 at 8 08 33 PM" src="https://github.com/user-attachments/assets/e986b0e1-52ef-4bcb-b4d1-3849810becba">

**Figure 4**. Restriction digest gel

Welp, that's not a good image for genotyping. Carolyn's getting a fresh Chelex sample sent, so I'll resume this quest of mine when we get that sample.

### Going forward

1. Troubleshoot genotyping
2. Clarify methods for average TTR analysis (reach out to Andy, Nic, or Megan?)
2. Individual-level TTR data analysis
3. Determine methods for comparing population responses
3. Troubleshoot lipid assay protocol
5. Conduct lipid assay for crabs of interest

{% if page.comments %}

<div id="disqus_thread"></div>
<script>

/**
*  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
*  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
/*
var disqus_config = function () {
this.page.url = PAGE_URL;  // Replace PAGE_URL with your page's canonical URL variable
this.page.identifier = PAGE_IDENTIFIER; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
};
*/
(function() { // DON'T EDIT BELOW THIS LINE
var d = document, s = d.createElement('script');
s.src = 'https://the-responsible-grad-student.disqus.com/embed.js';
s.setAttribute('data-timestamp', +new Date());
(d.head || d.body).appendChild(s);
})();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>

{% endif %}

<script id="dsq-count-scr" src="//the-responsible-grad-student.disqus.com/count.js" async></script>
